src/player_activity.cpp

```diff
@@ -132,6 +132,13 @@ void player_activity::do_turn( player &p )
     if( *this && type->will_refuel_fires() ) {
         try_fuel_fire( *this, p );
     }
+    if( *this && !p.is_npc() && type->valid_auto_needs() ) {
+        if( p.stomach.contains() <= p.stomach.capacity( p ) / 4 && p.get_kcal_percent() < 0.95f ) {
+            find_auto_consume( p, true );
+        } else if( p.get_thirst() > 130 ) {
+            find_auto_consume( p, false );
```

Comment:
`find_auto_consume` is called each turn. Will it cause performance problems if a large number of inedible foods/drinks is in the zone?

Reply:
I think what you mean is if there are many items, but none will quench thirst or satisfy hunger. It does seem like it should stash some kind of cooldown somewhere so it only checks occasionally if it finds nothing.

Reply:
I can add the cooldown in the activity object.